benchmark/FedHPOB/fedhpob/optimizers/grid_search.py

```diff
@@ -0,0 +1,49 @@
+# Implement GS with `ConfigSpace.util.generate_grid`
+
+import time
+import random
+import logging
+from ConfigSpace.util import generate_grid
+
+from fedhpob.config import fhb_cfg
+from fedhpob.utils.monitor import Monitor
+
+logging.basicConfig(level=logging.WARNING)
+
+
+def run_grid_search(cfg):
+    monitor = Monitor(cfg)
+    cfg = cfg.clone()
+    benchmark = cfg.benchmark.cls[0][cfg.benchmark.type](
+        cfg.benchmark.model,
+        cfg.benchmark.data,
+        cfg.benchmark.algo,
+        device=cfg.benchmark.device)
+    configspace = cfg.benchmark.configuration_space[0]
+    grid = generate_grid(configspace)
```

Comment:
Is it possible to specify the budget (the number of sampled points) in this step? Otherwise, we have no control about how dense the grids are.

Reply:
Since we have time budget control by `monitor`, we should set the number of sampled points manually (like interval each iter step), where 100/960 of FEMNIST configs are chosen in the gird search.

Reply:
> Is it possible to specify the budget (the number of sampled points) in this step? Otherwise, we have no control about how dense the grids are.

`ConfigSpace.util.generate_grid()`  won't work with Cate param, so I hard code the `configspace` of FEMNIST for exp usage.